feat: added implementation and configuration for issuer verification of jwt token. Added new attribute requiredIssuer to auth keys configuration. (#1030)
SDA-SE/sda-dropwizard-commons: sda-commons-server-auth/README.md

The authentication can be disabled for use in test and development environments. **Be careful to NEVER disable
authentication in production.**

For the authentication provider root URL and a jwks source a required issuer can be configured by the attribute `requiredIssuer`.
If the attribute `requiredIssuer` is set, the issuer of the token must match to the provided required issuer, and

> the host name of the source url mast be equal to the host name of the token issuer. 

Reviewer: typo: "must"

> If the attribute `requiredIssuer` is set, the issuer of the token must match to the provided required issuer, and
> the host name of the source url must be equal to the host name of the token issuer. 

Reviewer 2: ```suggestion
If the attribute `requiredIssuer` is set, the issuer of the token must match to the provided required issuer.
A warning will be logged if the host name of the source url does not match the host name of the required issuer.
```
